cmd/cosign/cli/env_test.go

```diff
@@ -16,22 +16,204 @@
 package cli
 
 import (
+	"io"
 	"os"
 	"testing"
 
-	"github.com/google/go-cmp/cmp"
+	"github.com/sigstore/cosign/pkg/cosign/env"
 )
 
-func TestGetEnv(t *testing.T) {
-	os.Setenv("COSIGN_EXPERIMENTAL", "foo")
-	os.Setenv("TUF_ROOT", "bar")
-	got := getEnv()
-	want := []string{
-		"COSIGN_EXPERIMENTAL=foo",
-		"TUF_ROOT=bar",
+const (
+	VariableTest1 env.Variable = "COSIGN_TEST1"
+	VariableTest2 env.Variable = "COSIGN_TEST2"
+
+	expectedWithoutDescription = `COSIGN_TEST1="abcd"
+COSIGN_TEST2=""
+`
+	expectedWithDescription = `# COSIGN_TEST1 is the first test variable
+# Expects: test1 value
+COSIGN_TEST1="abcd"
+# COSIGN_TEST2 is the second test variable
+# Expects: test2 value
+COSIGN_TEST2=""
+`
+
+	expectedWithHiddenSensitive = `# COSIGN_TEST1 is the first test variable
+# Expects: test1 value
+COSIGN_TEST1="abcd"
+# COSIGN_TEST2 is the second test variable
+# Expects: test2 value
+COSIGN_TEST2="******"
+`
+
+	expectedWithSensitive = `# COSIGN_TEST1 is the first test variable
+# Expects: test1 value
+COSIGN_TEST1="abcd"
+# COSIGN_TEST2 is the second test variable
+# Expects: test2 value
+COSIGN_TEST2="1234"
+`
+
+	expectedSensitiveWithoutDescription = `COSIGN_TEST1="abcd"
+COSIGN_TEST2="1234"
+`
+
+	expectedWithNonRegisteredEnv = `# COSIGN_TEST1 is the first test variable
+# Expects: test1 value
+COSIGN_TEST1="abcd"
+# COSIGN_TEST2 is the second test variable
+# Expects: test2 value
+COSIGN_TEST2=""
+# Environment variables below are not registered with cosign,
+# but might still influence cosign's behavior.
+COSIGN_TEST3=abcd
```

Comment:
Do we want quotes here? `COSIGN_TEST3="abcd"`

Reply:
Or get rid of quotes for the other variables. `env` doesn't show quotes, even if the variable has a space in it. 

```shell
$ env
env | head -n 3
TERM_SESSION_ID=w1t0p0:674E0F26-D173-4E6D-A95C-F26D8DEE667D
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.g6K1fscsWe/Listeners
LC_TERMINAL_VERSION=3.4.15
ITERM_PROFILE=Hotkey Window
```

Don't feel strongly either way but consistency is nice 🙂

Reply:
I also think consistency is nice. I removed quotes, it should be the same as for `env` and `os.Environ()` in Go.